usaspending_api/etl/management/commands/swap_in_new_table.py

```diff
@@ -0,0 +1,292 @@
+
+        if rename_sql:
+            cursor.execute("\n".join(rename_sql))
+
+    def swap_constraints_sql(self, cursor):
+        logging.info("Renaming constraints of the new and old tables.")
+        temp_constraints = self.query_result_lookup["temp_table_constraints"]
+        curr_constraints = self.query_result_lookup["curr_table_constraints"]
+        rename_sql = []
+        sql_template = "ALTER TABLE {table_name} RENAME CONSTRAINT {old_constraint_name} TO {new_constraint_name};"
+        for val in curr_constraints:
+            old_name = val["constraint_name"]
+            new_name = f"{old_name}_old"
+            rename_sql.append(
+                sql_template.format(
+                    table_name=self.db_table_name, old_constraint_name=old_name, new_constraint_name=new_name
+                )
+            )
+        for val in temp_constraints:
+            old_name = val["constraint_name"]
+            new_name = re.match("^(.*)_temp$", old_name, flags=re.I)[1]
+            rename_sql.append(
+                sql_template.format(
+                    table_name=self.temp_db_table_name, old_constraint_name=old_name, new_constraint_name=new_name
+                )
+            )
+
+        if rename_sql:
+            cursor.execute("\n".join(rename_sql))
+
+    @transaction.atomic
+    def swap_table_sql(self, cursor):
+        logging.info("Renaming the new and old tables")
+        sql_template = "ALTER TABLE {old_table_name} RENAME TO {new_table_name};"
+        rename_sql = [
+            sql_template.format(old_table_name=self.db_table_name, new_table_name=f"{self.db_table_name}_old"),
+            sql_template.format(old_table_name=self.temp_db_table_name, new_table_name=f"{self.db_table_name}"),
+        ]
+        cursor.execute("\n".join(rename_sql))
+
+    def drop_old_table_sql(self, cursor):
+        # Instead of using CASCADE, all old constraints and indexes are dropped manually
+        logging.info("Dropping the old table.")
+        drop_sql = []
+        indexes = self.query_result_lookup["curr_table_indexes"]
+        constraints = self.query_result_lookup["curr_table_constraints"]
+        for val in indexes:
+            name = f"{val['indexname']}_old"
+            drop_sql.append(f"DROP INDEX {name};")
+        for val in constraints:
+            name = f"{val['constraint_name']}_old"
+            drop_sql.append(f"ALTER TABLE {self.db_table_name}_old DROP CONSTRAINT {name};")
+        drop_sql.append(f"DROP TABLE {self.db_table_name}_old;")
+        cursor.execute("\n".join(drop_sql))
+
+    def extra_sql(self, cursor):
+        cursor.execute(f"ANALYZE VERBOSE {self.db_table_name}")
+        cursor.execute(f"GRANT SELECT ON {self.db_table_name} TO readonly")
+
+    def validate_tables(self, cursor):
+        logger.info("Verifying that the old table exists.")
+        cursor.execute(f"SELECT * FROM information_schema.tables WHERE table_name = '{self.db_table_name}'")
```

Comment:
Do we need to qualify the select with the schema of the target table and swap table